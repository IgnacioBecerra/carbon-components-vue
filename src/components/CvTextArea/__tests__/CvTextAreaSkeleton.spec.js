import { shallowMount } from '@vue/test-utils';
import { carbonPrefix } from '../../../global/settings';
import { CvTextAreaSkeleton } from '../';

describe('CvTextAreaSkeleton', () => {
  it('should match snapshot', async () => {
    const wrapper = await shallowMount(CvTextAreaSkeleton);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders text area skeleton', () => {
    const wrapper = shallowMount(CvTextAreaSkeleton);

    const skeletonLabel = wrapper.find(`.${carbonPrefix}--label`);
    const skeletonInput = wrapper.find(`.${carbonPrefix}--text-area`);
    const expectedSkeletonClass = `${carbonPrefix}--skeleton`;

    expect(skeletonLabel.classes()).toContain(expectedSkeletonClass);
    expect(skeletonInput.classes()).toContain(expectedSkeletonClass);
  });
});
