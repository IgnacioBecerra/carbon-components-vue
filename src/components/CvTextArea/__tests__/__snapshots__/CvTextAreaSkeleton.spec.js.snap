// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvTextAreaSkeleton should match snapshot 1`] = `
<div class="bx--form-item"><label class="bx--label bx--skeleton"></label>
  <div class="bx--text-area bx--skeleton"></div>
</div>
`;
